### ISSUE TITLE
Provide more info in `ChunkLoadError`

### ISSUE THREAD

**Issue by Joozty** [webpack/webpack#16618]
Would it be possible to provide more info in [ChunkLoadError](https://github.com/webpack/webpack/blob/main/lib/web/JsonpChunkLoadingRuntimeModule.js#L309), i.e. exact reason of failing?

I have an interesting case where from time to time my chunk is not loaded. When I open `devtools` I can see that request for chunk is successful but the chunk is still not loaded. I have implemented retry strategy for these cases but it seems it does not help. 

I would like to provide you with minimal working example but unfortunately I am not able to provide one. I think there is something interfering with the webpack code. Maybe some old library or something hence more info in the error or original event attached to the error would be nice to have. Maybe the original event contains more information about failing. 

As you can see from the screenshot the bundle loads correctly but it still throws ChunkLoadError hence I try it again 3x but it still fails.

![Screenshot 2023-01-06 at 15 57 48](https://user-images.githubusercontent.com/12775440/211039272-819e6efc-1d69-49ea-9d1d-dd1604091b72.png)

Would it be please possible to attach original event to Error the same way as [Error.request](https://github.com/webpack/webpack/blob/main/lib/web/JsonpChunkLoadingRuntimeModule.js#L182) is?

**Comment by Joozty** [comment]
I need to known while it failed. Is there any timeout for load? I did not known about that.

**Comment by alexander-akait** [comment]
Do you want to improve it and send a PR?

**Comment by Joozty** [comment]
Unfortunately no... I am not very familiar with the source code and build process.

**Comment by aespinolopez** [comment]
I faced recently a similar scenario: the error was caused by a syntax error in safari due to a lookbehind regex (not supported in safari).
The chunk was downloaded successfully but an exception was triggered during script evaluation and reported by webpack runtime as loading error.

I can share a reproducible example if needed and glad to take a look at it and send a PR!

**Comment by aespinolopez** [comment]
Hi @alexander-akait 
I created a [reproducible repo](https://github.com/aespinolopez/webpack_script_evaluation_error) for my comment above.

Both the parsing error and webpack chunk error are reported to console and the orignal error should always be displayed by the [specificatoin](https://html.spec.whatwg.org/multipage/webappapis.html#creating-scripts) if i understand it correctly.
The parsing error will be opaque in case of cross origin scripts and it will log Script Error.

In order to add the original error to ChunkLoadError a global error handler is needed and i don't know if this is a good idea (i'm open to suggestions but i think it is the only possible implementation) so my suggestion would be to change the error message of ChunkLoadError IF my assumptions are correct:

Webpack reports the error on **lib/web/JsonpChunkLoadingRuntimeModule.js:177**. This callback is attached to the script onload/onerror handlers and only runs if **installedChunks** is truthy (only possible truthy value is [res, rej, promise]) meaning the chunk has not load correctly.
In this scenario if the handler for the callback is onload the error type is set to **missing** and i think this is misleading and incorrect because the only possible scenario here is that the script has actually being loaded but an exception was thown during parsing / evaluation otherwise installedChunkData will be 0.

As conclusion i think the log  "missing: {url}" should be change to a value like ScriptError: {url}

**Comment by alexander-akait** [comment]
@aespinolopez sorry for delay, if you are intresting you can send a PR and I can help you

**Comment by aespinolopez** [comment]
@alexander-akait sure, glad to send a PR, but i don't know what is the expected solution. 

- 1. Changing the wording of the error message to reflect a runtime error during evaluation / execution
- 2. Looking for the real error message to report it in webpack's runtime
- some alternative solution

**Comment by alexander-akait** [comment]
@aespinolopez I think

> Looking for the real error message to report it in webpack's runtime

I think is better

**Comment by danantal** [comment]
Would also love to see this merged. We are facing some similar issues, where at least from the metrics perspective the number of errors we get is higher than the failed network requests - so we are a bit in the dark of what is going on.

Would be great if we can get some more information in our logs on the reason why chunks failed to load :)

**Comment by rtsscy** [comment]
> Would it be possible to provide more info in [ChunkLoadError](https://github.com/webpack/webpack/blob/main/lib/web/JsonpChunkLoadingRuntimeModule.js#L309), i.e. exact reason of failing?
> 
> I have an interesting case where from time to time my chunk is not loaded. When I open `devtools` I can see that request for chunk is successful but the chunk is still not loaded. I have implemented retry strategy for these cases but it seems it does not help.
> 
> I would like to provide you with minimal working example but unfortunately I am not able to provide one. I think there is something interfering with the webpack code. Maybe some old library or something hence more info in the error or original event attached to the error would be nice to have. Maybe the original event contains more information about failing.
> 
> As you can see from the screenshot the bundle loads correctly but it still throws ChunkLoadError hence I try it again 3x but it still fails.
> 
> ![Screenshot 2023-01-06 at 15 57 48](https://user-images.githubusercontent.com/12775440/211039272-819e6efc-1d69-49ea-9d1d-dd1604091b72.png)
> 
> Would it be please possible to attach original event to Error the same way as [Error.request](https://github.com/webpack/webpack/blob/main/lib/web/JsonpChunkLoadingRuntimeModule.js#L182) is?
Any chance you managed to figure out what is the issue? I am getting the exact same error, i can see the chunk in my devtools but somehow failing to load it :/

**Comment by rtsscy** [comment]
Any chance you managed to figure out what is the issue? I am getting the exact same error, i can see the chunk in my devtools but somehow failing to load it :/

**Comment by alexander-akait** [comment]
bump

**Comment by flucivja** [comment]
Hey folks, I encountered same issue in my app and found something interesting which could explain randomness.

- I have chunk CSS (`my-chunk.css`) which imports fonts from google like this `@import url('https://fonts.googleapis.com/css2?family=Inter:wght@100;200;300;400;500;600;700;800;900&display=swap');`
- and if I block `fonts.googleapis.com` domain, I get ChunkLoadError with information that `my-chunk.css` failed to be loaded even I see that it was loaded fine in network

And as far as I know, googleapis is blocked in some countries (e.g. China) or can be blocked by some firewalls or maybe even browser extensions because Google track user IP address.

**Comment by alexander-akait** [comment]
@flucivja hm sounds intresting, can you provide an example?

**Comment by flucivja** [comment]
@alexander-akait here is an example: https://codesandbox.io/p/sandbox/frosty-feynman-m79rwq.

But you must download code and run it locally because codesandbox does not apply MiniCssExtractPlugin for some reason.
When you run it locally, check network, there is font loading from googleapis which is imported in `Article.css` chunk.
Then block this request and refresh page (also attaching screenshots below with showing this behavior).

Before blocking request to googleapis:
![image](https://github.com/webpack/webpack/assets/4375072/7d43eccf-4859-4273-b0a3-c406be5f0846)

After blocking request to googleapis:
![image](https://github.com/webpack/webpack/assets/4375072/7d296ff8-a0f3-44a3-9dfa-102754abe864)

**Comment by alexander-akait** [comment]
@flucivja Sorry for the delay in response, yes I see the problem. An interesting problem is that if at least some module in the application/chunk is not available, then the application is broken, it is fully true for JS, because you code will be broken, but there are a cases for CSS like you, I think we need an option for such cases, and the developers must decide for themselves, because theoretically any code you have could be related to this, for example you CSS file can set custom properties and your JS logic can rely on such values

Can you open an issue in mini-css-extract-plugin, I will investigate it deeply and provide possible solution

Sorry for delay, again, a lot of issues

**Comment by alexander-akait** [comment]
Also I am thinking about:
```
import(/** webpackAllowFailureLoading: true */ "./Article.css");
```

**Comment by VAKHULA** [comment]
bump

**Comment by muskaan1212** [comment]
Hi everyone! I’m Muskaan, a CS student interested in contributing to webpack and learning more about its runtime and documentation ecosystem.
I recently started looking into `ChunkLoadError` handling around issue #16618 and opened a small exploratory PR. I’m still new to the codebase, so I’d appreciate any guidance on beginner-friendly areas where I can help, especially around developer experience, documentation, runtime error messages, or tests.
